.github/workflows/build_and_test.yml

```diff
@@ -297,6 +298,80 @@ jobs:
           fail_ci_if_error: false
           verbose: true
 
+  macos-m1:
+    runs-on: ${{ matrix.os-version }}
+    strategy:
+      matrix:
+        os-version: [macos-14]
+        arch: [arm64]
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - uses: actions/setup-python@v5
+        with:
+          # support version: https://raw.githubusercontent.com/actions/python-versions/main/versions-manifest.json
+          python-version: "3.12.x"
+
+      - name: Configure PcapPlusPlus
+        # Ensure user have access to network devices instead of giving super-user right
+        # Seems macos-14 has more strict file permission
+        run: |
+          sudo chmod a+rw /dev/bpf*
+          sudo chmod a+rw /usr/local
+          cmake -DCMAKE_OSX_ARCHITECTURES=${{ matrix.arch }} -DPCAPPP_BUILD_COVERAGE=ON -S . -B "$BUILD_DIR"
```

Comment:
As you're not cross compiling here no need to define CMAKE_OSX_ARCHITECTURES

Reply:
fixed.